File: httpreq.php
<?php
# Parse the HTTP Host header and request URI
# and write the data out in JSON format.

header('Content-type: application/json');
$hostname = $_SERVER['HTTP_HOST'];
$pathname = $_SERVER['REQUEST_URI'];

$data = array('hostname' => $hostname, 'pathname' => $pathname);

echo json_encode($data);
?>

File: httpreq.pl
#!/usr/bin/perl 

use strict;

sub quote {
  my $input = $_[0];
  $input =~ s/\\/\\\\"/g;
  $input =~ s/"/\\"/g;
  return $input;
}

my $hostname = &quote($ENV{HTTP_HOST});
my $pathname = &quote($ENV{REQUEST_URI});

#print "Content-Type: application/json; charset=utf-8\n";
print "Content-Type: application/javascript; charset=utf-8\n";
print "Access-Control-Allow-Origin: *\n\n";

#print "{\"hostname\":\"$hostname\",\"pathname\":\"$pathname\"}";

print "var hostname = \"$hostname\"; var pathname = \"$pathname\";";
